### PR TITLE
Issue #6 - Parse command line with docopt.

### PR DESCRIPTION
**Pull request by hfiguiere**

### CODE REVIEW

**Comment by hfiguiere** [comment]
@jedireza since you filed the issue, do you mind reviewing?

**Comment by hfiguiere** [comment]
It should be noted that PR #3 will instantly rot with this one. And vice versa.

**Comment by hfiguiere** [comment]
@fabricedesre ? Thanks !

File: src/main.rs [hunk 1]
```diff
@@ -23,6 +22,8 @@ extern crate router;
 extern crate serde;
 extern crate staticfile;
 extern crate uuid;
+extern crate docopt;
+extern crate rustc_serialize;
```
Comment (fabricedesre) [inline src/main.rs]: Why do we need that? It seems unused.

Comment (fabricedesre) [inline src/main.rs]: also, add them in alphabetical order

File: Cargo.toml [hunk 2]
```diff
@@ -17,3 +16,5 @@ serde_json = "0.6.0"
 serde_macros = "0.6.11"
 uuid = "0.1.18"
 clippy = "0.0.37"
+docopt = "0.6.78"
+rustc-serialize = "0.3"
```
Comment (fabricedesre) [inline Cargo.toml]: add the new dependencies in alphabetical order (bonus points if you also move clippy)

File: src/main.rs [hunk 3]
```diff
@@ -33,39 +34,38 @@ mod controler;
...
+const USAGE: &'static str = "
+Usage: foxbox [-v] [-h] [-n <hostname>]
+
+Options:
+    -v, --verbose            Toggle verbose output.
+    -n, --name               Set local hostname.
+    -h, --help               Print this help menu.
+";
+
+#[derive(RustcDecodable)]
```
Comment (jedireza) [inline src/main.rs:48]: Looks like this is what `rustc_serialize` was used for.

File: src/main.rs (hunk 1 above)
Comment (hfiguiere) [inline src/main.rs]: We need it for the Args. Will keep the crate sorted too.

File: src/main.rs (hunk 3 above)
Comment (hfiguiere) [inline src/main.rs:48]: Yup. That's the reason.

**Comment by hfiguiere** [comment]
I have reordered the dependencies as requested. And squished commits.

**Comment by hfiguiere** [comment]
@fabricedesre should be ready. Thanks !

**Pull request merged**